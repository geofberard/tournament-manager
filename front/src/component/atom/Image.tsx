import { Box, SxProps, Theme } from "@mui/material";
import * as React from "react";
import { FC } from "react";

export interface ImageProps {
    src: string;
    alt: string;
    sx: SxProps<Theme>;
}

export const Image: FC<ImageProps> = (props:ImageProps) => {
    return (
        <Box
            component="img"
            sx={props.sx}
            alt={props.alt}
            src={props.src}
        />
    );
};
