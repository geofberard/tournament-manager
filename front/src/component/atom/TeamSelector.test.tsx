import { SxProps, Theme } from '@mui/system';
import {waitFor, screen, getByRole} from '@testing-library/react'
import * as React from 'react';
import { render, fireEvent, testSnapshot } from '../../test/TestUtils';
import { Image } from "./Image"

testSnapshot(<Image src="url" alt="alt" sx={{color:"red"}}/>)

describe("initializing",() => {
    it('should use img tag', async () => {
        // When
        render(<Image />);
        
        // Then
        expect(screen.getByRole("img")).not.toBeUndefined();
    })

    it('should forward props to img tag', async () => {
        // Given
        const src:string = "imageSrc";
        const alt:string = "imageAlt";
        const sx:SxProps<Theme> = {background:"red"};
        
        // When
        render(<Image src={src} alt={alt} sx={sx} />);
        
        // Then
        const image = screen.getByRole("img");
        expect(image.getAttribute("src")).toBe(src);
        expect(image.getAttribute("alt")).toBe(alt);
        expect(window.getComputedStyle(image).background).toBe("red");
    })
})