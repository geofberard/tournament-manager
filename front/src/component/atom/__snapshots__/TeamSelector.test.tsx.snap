// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match last snapshot 1`] = `
<ForwardRef(Box)
  alt="alt"
  component="img"
  src="url"
  sx={
    Object {
      "color": "red",
    }
  }
/>
`;
